pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -56,9 +56,27 @@ type MatrixInclude struct {
 // Combination is a map, mainly defined to hold a single combination from a Matrix with key as param.Name and value as param.Value
 type Combination map[string]string
 
-// Combinations is a list of combinations
+// Combinations is a Combination list
 type Combinations []Combination
 
+// FanOut generates Combinations, which is a list of mapped Combination by fanning out matrix
+// Parameters including Matrix Include Parameters
+func (m *Matrix) FanOut() Combinations {
+	var combinations Combinations
+	// If only Matrix Include Parameters exists, generate and return explicit Combinations
+	if m.hasInclude() && !m.hasParams() {
+		return m.fanOutExplictCombinations()
+	}
+	// Fan out Matrix Parameters
+	for _, parameter := range m.Params {
+		combinations = combinations.fanOut(parameter)
+	}
+	// Replace initial Combinations generated with Matrix Include Parameters
+	mappedMatrixIncludeParamsSlice := m.extractIncludeParams()
+	combinations = combinations.replaceCombinations(mappedMatrixIncludeParamsSlice)
+	return combinations
+}
```

Comment:
https://github.com/tektoncd/pipeline/commit/586228bf97a23ab3b267a4a730bdd56867f2d771

Here ⬆️ is a reference implementation.

Reply:
all unit tests introduced in this PR succeeds using the reference implementation ⬆️

Reply:
Thanks Priti! I've incorporated these changes.